calico_containers/calico_ctl/node.py

```diff
@@ -109,8 +109,12 @@ def validate_arguments(arguments):
     asnum_ok = True
     if arguments.get("<AS_NUM>") or arguments.get("--as"):
         try:
-            asnum = int(arguments["<AS_NUM>"] or arguments["--as"])
-            asnum_ok = 0 <= asnum <= 4294967295
+            asnum = str(arguments.get("<AS_NUM>") or arguments.get("--as"))
+            if "." in asnum:
+                left_asnum, right_asnum = asnum.split(".")
+                asnum_ok = 0 <= (int(left_asnum) and int(right_asnum)) <= 65535
```

Comment:
This logic is not correct.

`int(left_asnum) and int(right_asnum)`

will return you int(right_asnum) if int(left_asnum) is non-zero, otherwise it'll return 0.  Which reduces your expression to:

asnum_ok = 0 <= int(right_asnum) <= 65535